logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputDelegatorExt.java

```diff
@@ -39,14 +40,17 @@ public OutputDelegatorExt initialize(final ThreadContext context, final RubyHash
         final ExecutionContextExt executionContext,
         final OutputStrategyExt.OutputStrategyRegistryExt strategyRegistry) {
         this.outputClass = outputClass;
-        initMetrics(
-            args.op_aref(context, RubyString.newString(context.runtime, "id")).asJavaString(),
-            metric
-        );
+
+        String id = args.op_aref(context, RubyString.newString(context.runtime, "id")).asJavaString();
+        RubyString configRefKey = RubyString.newString(context.runtime, "config-ref");
+        String configRef = args.op_aref(context, configRefKey).asJavaString();
+        initMetrics(id, metric, configRef);
+
+        args.remove(configRefKey);
```

Comment:
maybe instead of `args.op_aref()` and than `remove` do a: 
`IRubyObject configRef = args.delete(context, configRefKey)`
`nitMetrics(id, metric, configRef.asJavaString());`

NOTE: I am assuming the "config-ref" key is allways to be found in the map ...

Reply:
I don't know, it could be that value is null, in that case if I'm not wrong it compared as "null" string

Reply:
okay, let's not worry about that - we could still avoid doing a hash lookup twice 
for these 3 lines, doing something like: 
```java
IRubyObject configRef = args.delete(context, configRefKey)
nitMetrics(id, metric, configRef.asJavaString());
```
... do you feel its more confusing? (fine leaving it as is if you prefer)

Reply:
sounds good